C/common/storage_client.cpp

```diff
@@ -484,13 +484,26 @@ ReadingSet* StorageClient::queryTableToReadings(const std::string& tableName,
  * @return int		The number of rows inserted
  */
 int StorageClient::insertTable(const string& tableName, const InsertValues& values)
+{
+	return insertTable(DEFAULT_SCHEMA, tableName, values);
+}
+
+/**
+ * Insert data into an arbitrary table
+ *
+ * @param schema	The name of the schema to insert into
+ * @param tableName	The name of the table into which data will be added
+ * @param values	The values to insert into the table
+ * @return int		The number of rows inserted
+ */
+int StorageClient::insertTable(const string& schema, const string& tableName, const InsertValues& values)
 {
 	try {
 		ostringstream convert;
 
 		convert << values.toJSON();
 		char url[128];
-		snprintf(url, sizeof(url), "/storage/table/%s", tableName.c_str());
```

Comment:
Are we sure that no other client ( python client may be )  will ever try to post to  /storage/table/  and will always be posting to /storage/schema<schema_name><table><tablename> , if that is sure then there might  be changes in python client as well, storage service needs to be updated in a way so  as to support /storage/table url or not?

Reply:
> Are we sure that no other client ( python client may be ) will ever try to post to /storage/table/ and will always be posting to /storage/schema<schema_name>
> 
The is the C++ client, it doesn't matter what the Postgres client does to these code changes.



When we need the python code to use the new schema we will have to update the Python client.
> , if that is sure then there might be changes in python client as well, storage service needs to be updated in a way so as to support /storage/table url or not?

Yes, the service must support both URLs for backward compatibility. 

> What I mean to ask is that does /storage/table URL need to be supported by the storage service , if not then all the clients posting to /storage/table will have to be modified to past to new URL ( /storage/schema/.../table/...)